### PR TITLE
feat(FEC-12113): Add Side Panels Management to player UI

### PR DESCRIPTION
**Pull request by JonathanTGold**
### Description of the Changes

Add TypeScript types support according to last new typescript spec

Related pr: 
https://github.com/kaltura/playkit-js-ui-managers/pull/1
https://github.com/kaltura/playkit-js-ui/pull/674

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  };
+
+  SidePanelModes: {
+    ALONGSIDE: 'alongside',
+    HIDDEN: 'hidden',
+    OVER: 'over'
+  };
+
+  ReservedPresetNames: {
+    Playback: 'Playback',
```
Comment (yairans) [inline types.d.ts:59]: There are more

File: types.d.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+export interface Logger {
+  debug(message: any, ...optionalParams: any[]): void;
+  info(message: any, ...optionalParams: any[]): void;
+  trace(message: any, ...optionalParams: any[]): void;
+  warn(message: any, ...optionalParams: any[]): void;
+  error(message: any, ...optionalParams: any[]): void;
+}
+
+export interface PlaykitUI {
```
Comment (yairans) [inline types.d.ts:39]: possible to take from the UI itself? to avoid the duplication

**Review by yairans — CHANGES_REQUESTED** [review]

File: types.d.ts (hunk 2 above)
Comment (JonathanTGold) [reply types.d.ts:39]: the ui written in flow

File: types.d.ts (hunk 1 above)
Comment (JonathanTGold) [reply types.d.ts:59]: fixed

**Review by yairans — DISMISSED** [review]

**Review by yairans — APPROVED** [review]

**Pull request merged**